src/test/scala/com/microsoft/hyperspace/index/rankers/JoinIndexRankerTest.scala

```diff
@@ -31,57 +30,64 @@ class JoinIndexRankerTest extends SparkFunSuite {
 
     val expectedOrder = Seq((l_20, r_20), (l_10, r_20))
-    val actualOrder = JoinIndexRanker.rank(indexPairs)
+    val actualOrder = JoinIndexRanker.rank(indexPairs, false)
     assert(actualOrder.equals(expectedOrder))
   }
 
   test("testRank: Prefer higher number of buckets if multiple equal-bucket index pairs found") {
-    val l_10 = index("l1", Seq(t1c1), Seq(t1c2), 10)
-    val l_20 = index("l2", Seq(t1c1), Seq(t1c2), 20)
-    val r_10 = index("r1", Seq(t2c1), Seq(t2c2), 10)
-    val r_20 = index("r2", Seq(t2c1), Seq(t2c2), 20)
+    val l_10 = RankerTestHelper.index("l1", Seq(t1c1), Seq(t1c2), 10)
+    val l_20 = RankerTestHelper.index("l2", Seq(t1c1), Seq(t1c2), 20)
+    val r_10 = RankerTestHelper.index("r1", Seq(t2c1), Seq(t2c2), 10)
+    val r_20 = RankerTestHelper.index("r2", Seq(t2c1), Seq(t2c2), 20)
 
     val indexPairs = Seq((l_10, r_10), (l_10, r_20), (l_20, r_20))
 
     val expectedOrder = Seq((l_20, r_20), (l_10, r_10), (l_10, r_20))
-    val actualOrder = JoinIndexRanker.rank(indexPairs)
+    val actualOrder = JoinIndexRanker.rank(indexPairs, false)
     assert(actualOrder.equals(expectedOrder))
   }
 
-  private def index(
-      name: String,
-      indexCols: Seq[AttributeReference],
-      includedCols: Seq[AttributeReference],
-      numBuckets: Int): IndexLogEntry = {
-    val sourcePlanProperties = SparkPlan.Properties(
-      Seq(),
-      null,
-      null,
-      LogicalPlanFingerprint(
-        LogicalPlanFingerprint.Properties(Seq(Signature("signClass", "sign(plan)")))))
-
-    val entry = IndexLogEntry(
-      name,
-      CoveringIndex(
-        CoveringIndex.Properties(
-          CoveringIndex.Properties
-            .Columns(indexCols.map(_.name), includedCols.map(_.name)),
-          IndexLogEntry.schemaString(schemaFromAttributes(indexCols ++ includedCols: _*)),
-          numBuckets)),
-      Content(Directory(name)),
-      Source(SparkPlan(sourcePlanProperties)),
-      Map())
-    entry.state = Constants.States.ACTIVE
-    entry
-  }
+  test("testRank: Prefer the largest source file list if HybridScan is enabled") {
+    val fileList1 = Seq(FileInfo("a", 1, 1), FileInfo("b", 1, 1))
+    val fileList2 = Seq(FileInfo("c", 1, 1), FileInfo("d", 1, 1), FileInfo("e", 1, 1))
+    val l_10 = RankerTestHelper.index("l1", Seq(t1c1), Seq(t1c2), 10, fileList2)
+    val l_20 = RankerTestHelper.index("l2", Seq(t1c1), Seq(t1c2), 20, fileList1)
+    val r_10 = RankerTestHelper.index("r1", Seq(t2c1), Seq(t2c2), 10, fileList2)
+    val r_20 = RankerTestHelper.index("r2", Seq(t2c1), Seq(t2c2), 20, fileList1)
+
+    {
+      // test original algorithm
+      val indexPairs = Seq((l_10, r_10), (l_10, r_20), (l_20, r_20))
+      val expectedOrder = Seq((l_20, r_20), (l_10, r_10), (l_10, r_20))
+      val actualOrder = JoinIndexRanker.rank(indexPairs, hybridScanEnabled = false)
+      assert(actualOrder.equals(expectedOrder))
+    }
+
+    {
+      val indexPairs = Seq((l_10, r_10), (l_10, r_20), (l_20, r_20))
```

Comment:
nit: for consistency, please add a comment (similar to `test original algorithm` to this code block as well.